src/backend/libs-and-metadata.md

```diff
@@ -0,0 +1,192 @@
+format is specific to `rustc`, and may change over time. This file contains:
+
+* Object code, which is the result of code generation. This is used during
+  regular linking. There is a separate `.o` file for each [codegen unit]. The
+  codegen step can be skipped with the [`-C
+  linker-plugin-lto`][linker-plugin-lto] CLI option, which means each `.o`
+  file will only contain LLVM bitcode, which can be used when using LTO.
+* [LLVM bitcode], which is a binary representation of LLVM's intermediate
+  representation, which is embedded as a section in the `.o` files. This can
+  be used for [Link Time Optimization] (LTO). This can be removed with the
+  [`-C embed-bitcode=no`][embed-bitcode] CLI option to improve compile times
+  and reduce disk space if LTO is not needed.
+* `rustc` [metadata], in a file named `lib.rmeta`.
+* A symbol table, which is generally a list of symbols with offsets to the
+  object file that contain that symbol. This is pretty standard for archive
+  files.
+
+[archive file]: https://en.wikipedia.org/wiki/Ar_(Unix)
+[LLVM bitcode]: https://llvm.org/docs/BitCodeFormat.html
+[Link Time Optimization]: https://llvm.org/docs/LinkTimeOptimization.html
+[codegen unit]: ../backend/codegen.md
+[embed-bitcode]: https://doc.rust-lang.org/rustc/codegen-options/index.html#embed-bitcode
+[linker-plugin-lto]: https://doc.rust-lang.org/rustc/codegen-options/index.html#linker-plugin-lto
+
+### dylib
+
+A `dylib` is a platform-specific shared library. It includes the `rustc`
+[metadata] in a special section called `.rustc` in a compressed format.
+
+### rmeta
+
+An `rmeta` file is custom binary format that contains the [metadata] for the
+crate. This file can be used for fast "checks" of a project by skipping all
+code generation (as is done with `cargo check`), collecting enough information
+for documentation (as is done with `cargo doc`), or for
+[pipelining](#pipelining). This file is created if the
+[`--emit=metadata`][emit] CLI option is used.
+
+`rmeta` files do not support linking, since they do not contain compiled
+object files.
+
+[emit]: https://doc.rust-lang.org/rustc/command-line-arguments.html#option-emit
+
+## Metadata
+
+The metadata contains a wide swath of different elements. This guide will not
+go into detail of every field it contains. You are encouraged to browse the
+[`CrateRoot`] definition to get a sense of the different elements it contains.
+Everything about metadata encoding and decoding is in the [`rustc_metadata`]
+package.
+
+Here are a few highlights of things it contains:
+
+* The version of the `rustc` compiler. The compiler will refuse to load files
+  from any other version.
+* The [Strict Version Hash](#strict-version-hash) (SVH). This helps ensure the
+  correct dependency is loaded.
+* The [Crate Disambiguator](#crate-disambiguator). This is a hash used for
+  various things to disambiguate between different crates of the same name.
```

Comment:
```suggestion
  to disambiguate between different crates of the same name.
```

Reply:
(You'll need to delete the "for" on the previous line as well.)